src/core/views.py

```diff
@@ -69,10 +69,8 @@ def user_login(request):
     next_url = request.GET.get('next', '')
     if request.user.is_authenticated:
         messages.info(request, 'You are already logged in.')
-        if next_url:
-            return redirect(next_url)
-        else:
-            return redirect(reverse('website_index'))
+        return redirect(logic.get_post_auth_url(request))
+
```

Comment:
Consider my suggestions above, this could be `redirect(request.site_type.auth_redirect_url())`